app/src/main/java/io/appium/uiautomator2/handler/GetDeviceInfo.java

```diff
@@ -38,6 +43,50 @@ public GetDeviceInfo(String mappedUri) {
         super(mappedUri);
     }
 
+    private static JSONArray extractNetworkInfo(DeviceInfoHelper deviceInfoHelper) throws JSONException {
+        JSONArray result = new JSONArray();
+        for (Network network: deviceInfoHelper.getNetworks()) {
+            JSONObject resultItem = new JSONObject();
+            NetworkInfo networkInfo = deviceInfoHelper.extractInfo(network);
+            if (networkInfo != null) {
+                resultItem.put("type", networkInfo.getType());
+                resultItem.put("typeName", networkInfo.getTypeName());
+                resultItem.put("subtype", networkInfo.getSubtype());
+                resultItem.put("subtypeName", networkInfo.getSubtypeName());
+                resultItem.put("isConnected", networkInfo.isConnected());
+                resultItem.put("connectionState", networkInfo.getDetailedState().ordinal());
+                resultItem.put("extraInfo", formatNull(networkInfo.getExtraInfo()));
+                resultItem.put("isAvailable", networkInfo.isAvailable());
+                resultItem.put("isFailover", networkInfo.isFailover());
+                resultItem.put("isRoaming", networkInfo.isRoaming());
+            }
+
+            NetworkCapabilities networkCaps = deviceInfoHelper.extractCapabilities(network);
+            JSONObject caps = new JSONObject();
+            if (networkCaps != null) {
+                caps.put("transportTypes",
+                        getField("mTransportTypes", networkCaps));
+                caps.put("networkCapabilities",
+                        getField("mNetworkCapabilities", networkCaps));
+                caps.put("linkUpstreamBandwidthKbps",
+                        networkCaps.getLinkUpstreamBandwidthKbps());
+                caps.put("linkDownBandwidthKbps",
+                        networkCaps.getLinkDownstreamBandwidthKbps());
+                caps.put("signalStrength",
+                        getField("mSignalStrength", networkCaps));
+                caps.put("networkSpecifier",
+                        formatNull(getField("mNetworkSpecifier", networkCaps)));
+                caps.put("SSID", formatNull(getField("mSSID", networkCaps)));
+            }
+            resultItem.put("capabilities", formatNull(networkCaps == null ? null : caps));
+
+            if (networkCaps != null || networkInfo != null) {
```

Comment:
Could we check that `resultItem` is not empty?

`if (resultItem.length() > 0) { ... }`

Reply:
changed